blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -83,6 +84,40 @@ public Link getLink() {
         return runs;
     }
 
+    public static Iterable<BlueRun> findRuns(Job job, final Link parent, int start, int limit){
+        final List<BlueRun> runs = new ArrayList<>();
+        Iterable<Job> pipelines;
+        if(job != null){
+            pipelines = ImmutableList.of(job);
+        }else{
+            pipelines = Jenkins.getInstance().getItems(Job.class);
+        }
+
+        int skipCount=0;
+        int limitCount=0;
+        for (Job p : pipelines) {
+            RunList<? extends Run> runList = p.getBuilds();
```

Comment:
@vivek I still think we have a problem here. I created a job, did a few hundred runs on it and then tried loading the activity for that job. The limit was 26, but `getBuilds` is returning a `List` with all the jobs in it. If there are 1000s of jobs and Jenkins has just been restarted, this will result in every one of those Runs being loaded from disk, which seems wrong to me when you're only interested in the first 26.

Did you look at the [comment I left on the JIRA ticket](https://issues.jenkins-ci.org/browse/JENKINS-39661?focusedCommentId=277090&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-277090)? I did an experiment using `LazyBuildMixIn` and `RunMap`. See https://github.com/tfennelly/blueocean-plugin/tree/JENKINS-39661

I'll do another test here with a few 1000 runs and report back.